text/0006-architecture-review-process.md

```diff
@@ -0,0 +1,172 @@
+  * Find someone to **chair** the review:
+    - For a Roadmap Review this will be the *least* senior manager with the ability to address the problem in its entirety.
+    - For a Design Review this will be an experienced engineer in the problem domain outside of the team willing to put time into the problem. As engineers don’t review their own code they shouldn’t review their own designs. The principle is to get the most informed and least biased feedback possible.
+
+2. The team produces a **Review Packet** designed to answer the question identified in step 1. The production of the Review Packet is expected to be collaborative and iterative to ensure that the review focuses on the key questions and the team presents its best case. This is particularly true of a Roadmap Review where the consequences of the answer are likely to be more significant.
+The Review Packet includes:
+  * A lay summary of the problem space and the stakeholders which is focused on defining a shared language and identifying the key forces behind the problem
+  * A brief that explains what the team is proposing. This should read more like an encyclopedia entry than a marketing document. The audience is the people in the review; i.e. this should attempt to plug organizational documentation holes. The documentation process should not be more onerous than is required for the review. The brief should identify:
+    - the forces on the system
+    - the essential architecture behind the proposal
+    - scenarios that exercise the relevant forces against the proposed architecture
+    - how the proposal handles these scenarios
+  * A competitive analysis suggesting alternatives, costs, and opportunities
+
+3. The chair creates a list of questions to be discussed during the review. These questions should be:
+  * Open-ended
+  * Made available to the the team prior to the review so they can prepare answers
+
+4. The chair convenes a review meeting. The discussions of this meeting should be carefully minuted. The meeting comprises:
+  * The team making the proposal
+  * The chair
+  * Other senior engineers who will provide valuable input
+
+The review meeting should be a discussion of the issues, but should avoid feeling pressured to make a decision. The goal is to understand the issues raised by the question(s) and the background from the review packet, and to add to it wisdom from the people at the review. The meeting may decide to alter the questions and undergo a subsequent review. We should be very open about conversations that happen after the review meeting.
+
+The ideal review process scales well. The same basic system should work for a quick 2 person decision over the best way to design a certain feature, and for a critical organization-wide decision about a path to take. The term “team” is used above although we strongly recommend design reviews for smaller questions. If you feel yourself wondering if some design is best, it should be easy to perform a review.
```

Comment:
I find lightweight design reviews are well-embodied as (for example) GitHub PRs with no code attached. Then, if we decide to go ahead, we add code. Though this might not work as well in Bugzilla-centric projects since Bugzilla descriptions are immutable, making it hard to maintain a running summary and leading to Death By 1000 Comments.

Reply:
Similar - We've found that in the early stages, even a PR is too heavy-weight and we've been using GDocs until we're 80% there. The remaining 80% is done in a mix of PRs, and bugzilla meta bugs.